### ISSUE TITLE
Telepresence list does not pick up kubens namespace

### ISSUE THREAD

**Issue by petermant** [telepresenceio/telepresence#1928]
When running `telepresence list`, I would expect the namespace set by kubens to be picked up by telepresence. The series of commands below shows the impact of adding the -n namespace option, which shouldn't be necessary if the selected one was picked up.

I've just started a new root daemon, so run connect to launch the user daemon:
```
% telepresence connect
INFO[0000] No config found. Using default
Launching Telepresence User Daemon
Connected to context docker-desktop (https://kubernetes.docker.internal:6443)
```

Check that default is currently showing, before I do anything:
```
% kubens
ambassador
default     <-- in the console this is highlighted
kube-node-lease
kube-public
kube-system
pete
```

I want to use the 'pete' namespace, so select it:
```
% kubens pete
Context "docker-desktop" modified.
Active namespace is "pete".
```
```
% kubens
ambassador
default
kube-node-lease
kube-public
kube-system
pete    <-- in the console this is now highlighted
```

```
% kubectl config get-contexts
CURRENT   NAME             CLUSTER          AUTHINFO         NAMESPACE
*         docker-desktop   docker-desktop   docker-desktop   pete
```

So I'd expect it to list the services in pete, but I get `No Workloads`:
```
% telepresence list
INFO[0000] No config found. Using default
No Workloads (Deployments, StatefulSets, or ReplicaSets)
```

Whereas if I use -n then I get the full list:
```
% telepresence list -n pete
INFO[0000] No config found. Using default
service-name              : ready to intercept (traffic-agent not yet installed)
...and lots of others....
```

I can find a reference to this [in the v1 docs](https://www.telepresence.io/docs/v1/reference/connecting/) but not in the v2 docs - but presumably the behaviour should be the same?

**Comment by thallgren** [comment]
The problem is that the user-daemon (the connector) initializes itself from the kubeconfig during `telepresence connect` and then never reads the kubeconfig again after that. So in order to switch namespace to `pete`, the user must do:
```
telepresence quit
kubens pete
telepresence list
```
This is of course not ideal. Telepresence must discover when the kubeconfig is modified and act accordingly.

**Comment by efunk** [comment]
Resolved in Telepresence 2.4.1.
